src/fidesops/db/base_class.py

```diff
@@ -234,6 +234,10 @@ def delete_all(cls, db: Session) -> int:  # pylint: disable=W0622
         deleted_count = db.query(cls).delete()
         return deleted_count
 
+    def refresh_from_db(self, db: Session) -> FidesopsBase:
```

Comment:
I like this! I've recently queried for records that I just updated

Reply:
There's an intricacy here in the way DB sessions work that I'll explain.

The data in a DB session isn't guaranteed to be the same as in the database. In the event that another session mutates that same data that's currently held in another session, the data in that session becomes out of date. This manifests in our API tests because we have one session scoped to the test, and another that exists on the server within the API endpoint itself. What's happening is the test-scoped session creates data that the endpoint scoped session then mutates, and the test-scoped session no longer has accurate data. Without an effective refresh our assertions past this time would fail. This is why we also call `db.expunge(...)` before we call this function, to manually purge this object from the session so it's refreshed from the database.

There's a rabbit hole to go down here, that if we call `db.expunge(self)` within `refresh_from_db(...)` the following error is raised:

```
sqlalchemy.exc.InvalidRequestError: Instance <FidesopsUser at 0x7f5d4c70a640> is not present in this Session
```

which would imply the instance isn't part of the session at all, when if you call this same method within a `pdb.set_trace()`, the method works without issue.

Reply:
That's pretty interesting! So if `db.expunge(self)` isn't called `refresh_from_db(...)` won't get the most update to date data from the DB? I guess I assumed that the new query would pull in data directly from the DB instead of the session.